### ISSUE TITLE
Problem after reboot or hard reset: policy restart=always/on-failure with icc=false/true and linked containers

### ISSUE THREAD

**Issue by funcmike** [moby/moby#17611]
Issue related to (https://github.com/docker/docker/issues/16548):

Policy restart=always or on-failure with linked containers.
After reboot/shutdown or hard reset on next boot most of containers that use links are down. 
Cannot bring containers up again with manual run of: docker-compose up.

Steps to reproduce:
Run docker with --icc=false or --icc=true
Run docker-compose up (with linked containers and restart=always or on-failure policy)
Reboot VM or Hard reset VM.

VirtualMachine Parallels
Docker version 1.9.0-dev, build 38d3fc0

Linux ubuntu-linux 3.19.0-25-generic #26~14.04.1-Ubuntu SMP Fri Jul 24 21:16:20 UTC 2015 x86_64 x86_64 x86_64 GNU/Linux

Containers: 41
Images: 408
Server Version: 1.9.0-dev
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 490
 Dirperm1 Supported: true
Execution Driver: native-0.2
Logging Driver: json-file
Kernel Version: 3.19.0-25-generic
Operating System: Ubuntu 14.04.3 LTS
CPUs: 2
Total Memory: 3.856 GiB
Name: ubuntu-linux
ID: AJTQ:X463:RBTZ:VEOI:AKYY:WPJE:CN5A:SBG5:P44N:YU6C:N6M6:X5CS
Debug mode (server): true
 File Descriptors: 133
 Goroutines: 243
 System Time: 2015-11-02T16:53:46.083220139+01:00
 EventsListeners: 1
 Init SHA1:
 Init Path: /usr/bin/docker
 Docker Root Dir: /var/lib/docker
WARNING: No swap limit support

Reboot with icc=false with links (restart=always):

ESC[31mERROESC[0m[0003] failed to umount /var/lib/docker/containers/68d4ca427e5c162002b8bba2c2cd64402bf2dec905f1f30ac5e8150210f6d625/shm: invalid argument
ESC[31mERROESC[0m[0003] failed to umount /var/lib/docker/containers/68d4ca427e5c162002b8bba2c2cd64402bf2dec905f1f30ac5e8150210f6d625/mqueue: invalid argument
ESC[31mERROESC[0m[0003] 68d4ca427e5c162002b8bba2c2cd64402bf2dec905f1f30ac5e8150210f6d625: Failed to umount ipc filesystems: failed to cleanup ipc mounts:
ESC[31mERROESC[0m[0003] Failed to start container 68d4ca427e5c162002b8bba2c2cd64402bf2dec905f1f30ac5e8150210f6d625: linknotrunning: Cannot link to a non running container: /callex_clxDbNotelco_1 AS /callex_clxNotes_1/clxDbNotelco_1
ESC[31mERROESC[0m[0003] failed to umount /var/lib/docker/containers/79bfb76a9c933d5e98538be6c816656cfefda7b643b894020bf57c7f0b446a08/shm: invalid argument
ESC[31mERROESC[0m[0003] failed to umount /var/lib/docker/containers/79bfb76a9c933d5e98538be6c816656cfefda7b643b894020bf57c7f0b446a08/mqueue: invalid argument
ESC[31mERROESC[0m[0003] 79bfb76a9c933d5e98538be6c816656cfefda7b643b894020bf57c7f0b446a08: Failed to umount ipc filesystems: failed to cleanup ipc mounts:
ESC[31mERROESC[0m[0003] Failed to start container 79bfb76a9c933d5e98538be6c816656cfefda7b643b894020bf57c7f0b446a08: invalid endpoint id: cee6d0503e0170e47b28dbd9f71d358459b5b8cb1027770385b421567e3a02c4
ESC[31mERROESC[0m[0010] failed to umount /var/lib/docker/containers/9511f721b7ba3d6032c3aacadb60d122f1f5d1a9e5678c694d3697287378020f/shm: invalid argument
ESC[31mERROESC[0m[0010] failed to umount /var/lib/docker/containers/9511f721b7ba3d6032c3aacadb60d122f1f5d1a9e5678c694d3697287378020f/mqueue: invalid argument
ESC[31mERROESC[0m[0010] 9511f721b7ba3d6032c3aacadb60d122f1f5d1a9e5678c694d3697287378020f: Failed to umount ipc filesystems: failed to cleanup ipc mounts:
ESC[31mERROESC[0m[0010] Failed to start container 9511f721b7ba3d6032c3aacadb60d122f1f5d1a9e5678c694d3697287378020f: linknotrunning: Cannot link to a non running container: /callex_clxDbNotelco_1 AS /callex_clxWWW_1/clxDbNotelco
ESC[31mERROESC[0m[0012] failed to umount /var/lib/docker/containers/e271e1a15f84667385ab908edb23e63a98044a2978b722738c1a28b67432cb23/shm: invalid argument
ESC[31mERROESC[0m[0012] failed to umount /var/lib/docker/containers/e271e1a15f84667385ab908edb23e63a98044a2978b722738c1a28b67432cb23/mqueue: invalid argument
ESC[31mERROESC[0m[0012] e271e1a15f84667385ab908edb23e63a98044a2978b722738c1a28b67432cb23: Failed to umount ipc filesystems: failed to cleanup ipc mounts:
ESC[31mERROESC[0m[0012] Failed to start container e271e1a15f84667385ab908edb23e63a98044a2978b722738c1a28b67432cb23: linknotrunning: Cannot link to a non running container: /callex_clxDbNotelco_1 AS /callex_provexApi_1/callex_clxDbNotelco_1
ESC[31mERROESC[0m[0021] failed to umount /var/lib/docker/containers/526ba44b3be97906bfe1274aeef9d70f21d6caa130e8c23cb466bdd0b7398424/shm: invalid argument
ESC[31mERROESC[0m[0021] failed to umount /var/lib/docker/containers/526ba44b3be97906bfe1274aeef9d70f21d6caa130e8c23cb466bdd0b7398424/mqueue: invalid argument
ESC[31mERROESC[0m[0021] 526ba44b3be97906bfe1274aeef9d70f21d6caa130e8c23cb466bdd0b7398424: Failed to umount ipc filesystems: failed to cleanup ipc mounts:
ESC[31mERROESC[0m[0021] Failed to start container 526ba44b3be97906bfe1274aeef9d70f21d6caa130e8c23cb466bdd0b7398424: linknotrunning: Cannot link to a non running container: /callex_provexApi_1 AS /callex_clxLbnoTelco_1/provexApi_1

after this:
Docker-compose up -d
Cannot start container ecaf41a9ef4c78432e95c7c87b956c331d25c05fd6fa2f247c1f543528204759: container already connected to network bridge

Reboot with icc=true with links (restart=always):

ESC[31mERROESC[0m[0000] getNetworkFromStore for nid 8386ee5c6bb3e40e445d0e64aa9fb50a9933e14d39384442e1e2021ec55f670c failed while trying to build sandbox for cleanup: network 8386ee5c6bb3e40e445d0e64aa9fb50a9933e14d39384442e1e2021ec55f670c not found
ESC[31mERROESC[0m[0000] failed to delete sandbox 26bae7ca4bf4f75f96de3a987b829ff82f5003e33d0bd8ea28acf98da559bced while trying to cleanup: could not cleanup all the endpoints in container def48c66ed8bb5763a21b11174c2190faca629e6c788f70539eb42207330dd4d / sandbox 26bae7ca4bf4f75f96de3a987b829ff82f5003e33d0bd8ea28acf98da559bced
ESC[31mERROESC[0m[0000] getNetworkFromStore for nid 8386ee5c6bb3e40e445d0e64aa9fb50a9933e14d39384442e1e2021ec55f670c failed while trying to build sandbox for cleanup: network 8386ee5c6bb3e40e445d0e64aa9fb50a9933e14d39384442e1e2021ec55f670c not found
ESC[31mERROESC[0m[0000] failed to delete sandbox f22a7ca4a44a8ccf939326de02adcf4c1754c84f390d8eafd1fde75adbf58b4d while trying to cleanup: could not cleanup all the endpoints in container 39cb12f4aa9e8b3f41b80f26a3b393219c35f9192b591e1c371ba9ffa320f407 / sandbox f22a7ca4a44a8ccf939326de02adcf4c1754c84f390d8eafd1fde75adbf58b4d
ESC[31mERROESC[0m[0001] failed to umount /var/lib/docker/containers/22be593a3a7d0e38a829016ae0b75a3944f596957be754302ce554316f1e8d20/shm: invalid argument
ESC[31mERROESC[0m[0001] failed to umount /var/lib/docker/containers/22be593a3a7d0e38a829016ae0b75a3944f596957be754302ce554316f1e8d20/mqueue: invalid argument
ESC[31mERROESC[0m[0001] 22be593a3a7d0e38a829016ae0b75a3944f596957be754302ce554316f1e8d20: Failed to umount ipc filesystems: failed to cleanup ipc mounts:
ESC[31mERROESC[0m[0001] Failed to start container 22be593a3a7d0e38a829016ae0b75a3944f596957be754302ce554316f1e8d20: nosuchcontainer: no such id: 5063fbe531e2a2044fc0cad4d47cf52ca6414dea95a15b6f64cdf4647e3aa841
ESC[31mERROESC[0m[0002] failed to umount /var/lib/docker/containers/7c187b6bc4fdb302135985caca468b23aaf2628a09e17b66a5f830878f08052f/shm: invalid argument
ESC[31mERROESC[0m[0002] failed to umount /var/lib/docker/containers/7c187b6bc4fdb302135985caca468b23aaf2628a09e17b66a5f830878f08052f/mqueue: invalid argument
ESC[31mERROESC[0m[0002] 7c187b6bc4fdb302135985caca468b23aaf2628a09e17b66a5f830878f08052f: Failed to umount ipc filesystems: failed to cleanup ipc mounts:
ESC[31mERROESC[0m[0002] Failed to start container 7c187b6bc4fdb302135985caca468b23aaf2628a09e17b66a5f830878f08052f: linknotrunning: Cannot link to a non running container: /callex_clxDbNotelco_1 AS /callex_provexApi_1/clxDbNotelco
ESC[31mERROESC[0m[0002] failed to umount /var/lib/docker/containers/0c9682defa339263b99cff95aa4c433eaa0eabe63df7cd12fe1ce58258fd897e/shm: invalid argument
ESC[31mERROESC[0m[0002] failed to umount /var/lib/docker/containers/0c9682defa339263b99cff95aa4c433eaa0eabe63df7cd12fe1ce58258fd897e/mqueue: invalid argument
ESC[31mERROESC[0m[0002] 0c9682defa339263b99cff95aa4c433eaa0eabe63df7cd12fe1ce58258fd897e: Failed to umount ipc filesystems: failed to cleanup ipc mounts:
ESC[31mERROESC[0m[0002] Failed to start container 0c9682defa339263b99cff95aa4c433eaa0eabe63df7cd12fe1ce58258fd897e: linknotrunning: Cannot link to a non running container: /callex_clxDbNotelco_1 AS /callex_clxNotes_1/clxDbNotelco
ESC[31mERROESC[0m[0003] failed to umount /var/lib/docker/containers/e1cfc3cbb02ff140e5014bb96be0ae463e98bc4fb3539b195f17a999a89d811f/shm: invalid argument
ESC[31mERROESC[0m[0003] failed to umount /var/lib/docker/containers/e1cfc3cbb02ff140e5014bb96be0ae463e98bc4fb3539b195f17a999a89d811f/mqueue: invalid argument
ESC[31mERROESC[0m[0003] e1cfc3cbb02ff140e5014bb96be0ae463e98bc4fb3539b195f17a999a89d811f: Failed to umount ipc filesystems: failed to cleanup ipc mounts:
ESC[31mERROESC[0m[0003] Failed to start container e1cfc3cbb02ff140e5014bb96be0ae463e98bc4fb3539b195f17a999a89d811f: linknotrunning: Cannot link to a non running container: /callex_clxNotes_1 AS /callex_clxAgentPanel_1/clxNotes
ESC[31mERROESC[0m[0005] failed to umount /var/lib/docker/containers/b06e979c33e9ce7e89180f9149b3e53f12b05b07bad62f17608e2174d7143528/shm: invalid argument
ESC[31mERROESC[0m[0005] failed to umount /var/lib/docker/containers/b06e979c33e9ce7e89180f9149b3e53f12b05b07bad62f17608e2174d7143528/mqueue: invalid argument
ESC[31mERROESC[0m[0005] b06e979c33e9ce7e89180f9149b3e53f12b05b07bad62f17608e2174d7143528: Failed to umount ipc filesystems: failed to cleanup ipc mounts:
ESC[31mERROESC[0m[0005] Failed to start container b06e979c33e9ce7e89180f9149b3e53f12b05b07bad62f17608e2174d7143528: linknotrunning: Cannot link to a non running container: /callex_provexApi_1 AS /callex_clxWWW_1/callex_provexApi_1
ESC[31mERROESC[0m[0005] failed to umount /var/lib/docker/containers/5234cbc1d2a07f77d131c22e94e10b196c5b32fcad239c7d6c8bcc420fa65b2d/shm: invalid argument
ESC[31mERROESC[0m[0005] failed to umount /var/lib/docker/containers/5234cbc1d2a07f77d131c22e94e10b196c5b32fcad239c7d6c8bcc420fa65b2d/mqueue: invalid argument
ESC[31mERROESC[0m[0005] 5234cbc1d2a07f77d131c22e94e10b196c5b32fcad239c7d6c8bcc420fa65b2d: Failed to umount ipc filesystems: failed to cleanup ipc mounts:
ESC[31mERROESC[0m[0005] Failed to start container 5234cbc1d2a07f77d131c22e94e10b196c5b32fcad239c7d6c8bcc420fa65b2d: linknotrunning: Cannot link to a non running container: /callex_clxWWW_1 AS /callex_clxLbnoTelco_1/clxWWW

After:
manual run of docker-compose up -d

Cannot start container 22be593a3a7d0e38a829016ae0b75a3944f596957be754302ce554316f1e8d20: container already connected to network bridge

Reboot with icc=true without links (restart=always):

ESC[31mERROESC[0m[0000] getNetworkFromStore for nid 8386ee5c6bb3e40e445d0e64aa9fb50a9933e14d39384442e1e2021ec55f670c failed while trying to build sandbox for cleanup: network 8386ee5c6bb3e40e445d0e64aa9fb50a9933e14d39384442e1e2021ec55f670c not found
ESC[31mERROESC[0m[0000] failed to delete sandbox 26bae7ca4bf4f75f96de3a987b829ff82f5003e33d0bd8ea28acf98da559bced while trying to cleanup: could not cleanup all the endpoints in container def48c66ed8bb5763a21b11174c2190faca629e6c788f70539eb42207330dd4d / sandbox 26bae7ca4bf4f75f96de3a987b829ff82f5003e33d0bd8ea28acf98da559bced
ESC[31mERROESC[0m[0000] getNetworkFromStore for nid 8386ee5c6bb3e40e445d0e64aa9fb50a9933e14d39384442e1e2021ec55f670c failed while trying to build sandbox for cleanup: network 8386ee5c6bb3e40e445d0e64aa9fb50a9933e14d39384442e1e2021ec55f670c not found
ESC[31mERROESC[0m[0000] failed to delete sandbox f22a7ca4a44a8ccf939326de02adcf4c1754c84f390d8eafd1fde75adbf58b4d while trying to cleanup: could not cleanup all the endpoints in container 39cb12f4aa9e8b3f41b80f26a3b393219c35f9192b591e1c371ba9ffa320f407 / sandbox f22a7ca4a44a8ccf939326de02adcf4c1754c84f390d8eafd1fde75adbf58b4d
ESC[31mERROESC[0m[0002] failed to umount /var/lib/docker/containers/5d280886bdb7b04d386104bc482e6a25139d76b887c3d23e79a64f34d58e8c1b/shm: invalid argument
ESC[31mERROESC[0m[0002] failed to umount /var/lib/docker/containers/5d280886bdb7b04d386104bc482e6a25139d76b887c3d23e79a64f34d58e8c1b/mqueue: invalid argument
ESC[31mERROESC[0m[0002] 5d280886bdb7b04d386104bc482e6a25139d76b887c3d23e79a64f34d58e8c1b: Failed to umount ipc filesystems: failed to cleanup ipc mounts:
ESC[31mERROESC[0m[0002] Failed to start container 5d280886bdb7b04d386104bc482e6a25139d76b887c3d23e79a64f34d58e8c1b: [8] System error: invalid argument

Hard reset (restart=on-failure):

ESC[31mERROESC[0m[0001] getNetworkFromStore for nid b43b742782192264a37b4a56b387784b137ce3f98e01b791cd555b53df6f3fcf failed while trying to build sandbox for cleanup: network b43b742782192264a37b4a56b38
7784b137ce3f98e01b791cd555b53df6f3fcf not found
ESC[31mERROESC[0m[0001] failed to delete sandbox 94e581ca6d5e7fe7e1c8a8770e0186839af1eead72f89b53a8332bde9ee2eff5 while trying to cleanup: could not cleanup all the endpoints in container b9a9dbc604fc36
7c781c538f327f9e5b5f86cb863dab0927e41ab9d47ef80e34 / sandbox 94e581ca6d5e7fe7e1c8a8770e0186839af1eead72f89b53a8332bde9ee2eff5
ESC[31mERROESC[0m[0004] getNetworkFromStore for nid b43b742782192264a37b4a56b387784b137ce3f98e01b791cd555b53df6f3fcf failed while trying to build sandbox for cleanup: network b43b742782192264a37b4a56b38
7784b137ce3f98e01b791cd555b53df6f3fcf not found
ESC[31mERROESC[0m[0004] failed to delete sandbox 67d282b28fdab4467f84e7109d6484706d5228dc7e59a82d13bb41d248ee6738 while trying to cleanup: could not cleanup all the endpoints in container 560a5995d424233b42b1038ac56dd766daa2e6178118be37b43eae501d3cd437 / sandbox 67d282b28fdab4467f84e7109d6484706d5228dc7e59a82d13bb41d248ee6738
ESC[31mERROESC[0m[0007] getNetworkFromStore for nid 8386ee5c6bb3e40e445d0e64aa9fb50a9933e14d39384442e1e2021ec55f670c failed while trying to build sandbox for cleanup: network 8386ee5c6bb3e40e445d0e64aa9fb50a9933e14d39384442e1e2021ec55f670c not found
ESC[31mERROESC[0m[0007] failed to delete sandbox f22a7ca4a44a8ccf939326de02adcf4c1754c84f390d8eafd1fde75adbf58b4d while trying to cleanup: could not cleanup all the endpoints in container 39cb12f4aa9e8b3f41b80f26a3b393219c35f9192b591e1c371ba9ffa320f407 / sandbox f22a7ca4a44a8ccf939326de02adcf4c1754c84f390d8eafd1fde75adbf58b4d
ESC[31mERROESC[0m[0008] getNetworkFromStore for nid b43b742782192264a37b4a56b387784b137ce3f98e01b791cd555b53df6f3fcf failed while trying to build sandbox for cleanup: network b43b742782192264a37b4a56b387784b137ce3f98e01b791cd555b53df6f3fcf not found
ESC[31mERROESC[0m[0008] failed to delete sandbox 1f1d5cc203108e2ae11395ecdbee79b12cf41f2d9f649b6e487b4777c4275180 while trying to cleanup: could not cleanup all the endpoints in container 89994ecb5802064c91542c47798ab3069a086bfd33880032ab59db09dc4ff33f / sandbox 1f1d5cc203108e2ae11395ecdbee79b12cf41f2d9f649b6e487b4777c4275180
ESC[31mERROESC[0m[0010] getNetworkFromStore for nid 8386ee5c6bb3e40e445d0e64aa9fb50a9933e14d39384442e1e2021ec55f670c failed while trying to build sandbox for cleanup: network 8386ee5c6bb3e40e445d0e64aa9fb50a9933e14d39384442e1e2021ec55f670c not found
ESC[31mERROESC[0m[0010] failed to delete sandbox 26bae7ca4bf4f75f96de3a987b829ff82f5003e33d0bd8ea28acf98da559bced while trying to cleanup: could not cleanup all the endpoints in container def48c66ed8bb5763a21b11174c2190faca629e6c788f70539eb42207330dd4d / sandbox 26bae7ca4bf4f75f96de3a987b829ff82f5003e33d0bd8ea28acf98da559bced
ESC[31mERROESC[0m[0012] nosuchcontainer: no such id: a08f057e92c078427ff7c911f4d97848a71af03e2f3334091c0bab71e3cdb71f
ESC[31mERROESC[0m[0012] nosuchcontainer: no such id: a08f057e92c078427ff7c911f4d97848a71af03e2f3334091c0bab71e3cdb71f
ESC[31mERROESC[0m[0012] nosuchcontainer: no such id: a08f057e92c078427ff7c911f4d97848a71af03e2f3334091c0bab71e3cdb71f
ESC[31mERROESC[0m[0012] failed to umount /var/lib/docker/containers/ed754fe0898f6f5a842720825df5ee47589dad98bda2a61596e8b25e6b2da40e/shm: invalid argument
ESC[31mERROESC[0m[0012] failed to umount /var/lib/docker/containers/ed754fe0898f6f5a842720825df5ee47589dad98bda2a61596e8b25e6b2da40e/mqueue: invalid argument
ESC[31mERROESC[0m[0012] ed754fe0898f6f5a842720825df5ee47589dad98bda2a61596e8b25e6b2da40e: Failed to umount ipc filesystems: failed to cleanup ipc mounts:
ESC[31mERROESC[0m[0012] Failed to start container ed754fe0898f6f5a842720825df5ee47589dad98bda2a61596e8b25e6b2da40e: linknotrunning: Cannot link to a non running container: /callex_centralLog_1 AS /callex_clxNotes_1/callex_centralLog_1
ESC[31mERROESC[0m[0012] failed to cleanup up stale network sandbox for container 89994ecb5802064c91542c47798ab3069a086bfd33880032ab59db09dc4ff33f
ESC[31mERROESC[0m[0012] Error deleting sandbox id 1f1d5cc203108e2ae11395ecdbee79b12cf41f2d9f649b6e487b4777c4275180 for container 89994ecb5802064c91542c47798ab3069a086bfd33880032ab59db09dc4ff33f: could not cleanup all the endpoints in container 89994ecb5802064c91542c47798ab3069a086bfd33880032ab59db09dc4ff33f / sandbox 1f1d5cc203108e2ae11395ecdbee79b12cf41f2d9f649b6e487b4777c4275180
ESC[31mERROESC[0m[0012] failed to umount /var/lib/docker/containers/89994ecb5802064c91542c47798ab3069a086bfd33880032ab59db09dc4ff33f/shm: invalid argument
ESC[31mERROESC[0m[0012] failed to umount /var/lib/docker/containers/89994ecb5802064c91542c47798ab3069a086bfd33880032ab59db09dc4ff33f/mqueue: invalid argument
ESC[31mERROESC[0m[0012] 89994ecb5802064c91542c47798ab3069a086bfd33880032ab59db09dc4ff33f: Failed to umount ipc filesystems: failed to cleanup ipc mounts:
ESC[31mERROESC[0m[0012] Failed to start container 89994ecb5802064c91542c47798ab3069a086bfd33880032ab59db09dc4ff33f: open : no such file or directory
ESC[31mERROESC[0m[0012] failed to cleanup up stale network sandbox for container b9a9dbc604fc367c781c538f327f9e5b5f86cb863dab0927e41ab9d47ef80e34
ESC[31mERROESC[0m[0012] Error deleting sandbox id 94e581ca6d5e7fe7e1c8a8770e0186839af1eead72f89b53a8332bde9ee2eff5 for container b9a9dbc604fc367c781c538f327f9e5b5f86cb863dab0927e41ab9d47ef80e34: could not cleanup all the endpoints in container b9a9dbc604fc367c781c538f327f9e5b5f86cb863dab0927e41ab9d47ef80e34 / sandbox 94e581ca6d5e7fe7e1c8a8770e0186839af1eead72f89b53a8332bde9ee2eff5
ESC[31mERROESC[0m[0012] failed to umount /var/lib/docker/containers/b9a9dbc604fc367c781c538f327f9e5b5f86cb863dab0927e41ab9d47ef80e34/shm: invalid argument
ESC[31mERROESC[0m[0012] failed to umount /var/lib/docker/containers/b9a9dbc604fc367c781c538f327f9e5b5f86cb863dab0927e41ab9d47ef80e34/mqueue: invalid argument
ESC[31mERROESC[0m[0012] b9a9dbc604fc367c781c538f327f9e5b5f86cb863dab0927e41ab9d47ef80e34: Failed to umount ipc filesystems: failed to cleanup ipc mounts:
ESC[31mERROESC[0m[0012] Failed to start container b9a9dbc604fc367c781c538f327f9e5b5f86cb863dab0927e41ab9d47ef80e34: open : no such file or directory
ESC[31mERROESC[0m[0012] failed to cleanup up stale network sandbox for container 560a5995d424233b42b1038ac56dd766daa2e6178118be37b43eae501d3cd437
ESC[31mERROESC[0m[0013] failed to umount /var/lib/docker/containers/0089cad02acd5c3d457c79136b7d5e3ba6dcb8391c00c54fdee7e1a6095dc73f/mqueue: invalid argument
ESC[31mERROESC[0m[0013] 0089cad02acd5c3d457c79136b7d5e3ba6dcb8391c00c54fdee7e1a6095dc73f: Failed to umount ipc filesystems: failed to cleanup ipc mounts:
ESC[31mERROESC[0m[0013] Failed to start container 0089cad02acd5c3d457c79136b7d5e3ba6dcb8391c00c54fdee7e1a6095dc73f: linknotrunning: Cannot link to a non running container: /callex_clxDbNotelco_1 AS /callex_provexApi_1/callex_clxDbNotelco_1
ESC[31mERROESC[0m[0014] failed to umount /var/lib/docker/containers/b2d409b53ae6217371b64e90210fc72a6e2746eb453b412179647855cfd8da12/shm: invalid argument
ESC[31mERROESC[0m[0014] failed to umount /var/lib/docker/containers/b2d409b53ae6217371b64e90210fc72a6e2746eb453b412179647855cfd8da12/mqueue: invalid argument
ESC[31mERROESC[0m[0014] b2d409b53ae6217371b64e90210fc72a6e2746eb453b412179647855cfd8da12: Failed to umount ipc filesystems: failed to cleanup ipc mounts:
ESC[31mERROESC[0m[0014] Failed to start container b2d409b53ae6217371b64e90210fc72a6e2746eb453b412179647855cfd8da12: linknotrunning: Cannot link to a non running container: /callex_clxDbNotelco_1 AS /callex_clxWWW_1/clxDbNotelco
ESC[31mERROESC[0m[0015] failed to umount /var/lib/docker/containers/4b4e5d6b1b1aa13acbe32e0ee9e4e307db039424e3428e550047be743be65f6f/shm: invalid argument
ESC[31mERROESC[0m[0015] failed to umount /var/lib/docker/containers/4b4e5d6b1b1aa13acbe32e0ee9e4e307db039424e3428e550047be743be65f6f/mqueue: invalid argument
ESC[31mERROESC[0m[0015] 4b4e5d6b1b1aa13acbe32e0ee9e4e307db039424e3428e550047be743be65f6f: Failed to umount ipc filesystems: failed to cleanup ipc mounts:
ESC[31mERROESC[0m[0015] Failed to start container 4b4e5d6b1b1aa13acbe32e0ee9e4e307db039424e3428e550047be743be65f6f: linknotrunning: Cannot link to a non running container: /callex_clxWWW_1 AS /callex_clxLbnoTelco_1/clxWWW_1
ESC[31mERROESC[0m[0058] failed to cleanup up stale network sandbox for container 89994ecb5802064c91542c47798ab3069a086bfd33880032ab59db09dc4ff33f
ESC[31mERROESC[0m[0058] failed to umount /var/lib/docker/containers/89994ecb5802064c91542c47798ab3069a086bfd33880032ab59db09dc4ff33f/shm: invalid argument
ESC[31mERROESC[0m[0058] failed to umount /var/lib/docker/containers/89994ecb5802064c91542c47798ab3069a086bfd33880032ab59db09dc4ff33f/mqueue: invalid argument
ESC[31mERROESC[0m[0058] 89994ecb5802064c91542c47798ab3069a086bfd33880032ab59db09dc4ff33f: Failed to umount ipc filesystems: failed to cleanup ipc mounts:

After
manual run of docker-compose up -d

Cannot start container 89994ecb5802064c91542c47798ab3069a086bfd33880032ab59db09dc4ff33f: container already connected to network host

**Comment by cpuguy83** [comment]
I see you are running 1.9.0-dev... how old is this? Can you try with 1.9.0-rc4?

**Comment by mikechris** [comment]
Docker version 1.9.0-rc4, build e6f5a3c

Problem with automatic start of linked containers after reboot still exists.
Problem with "docker-compose up" after reboot is fixed - it's possible to manually start containers.

reboot with restart=always icc=false and links

ESC[31mERROESC[0m[0001] getNetworkFromStore for nid b43b742782192264a37b4a56b387784b137ce3f98e01b791cd555b53df6f3fcf failed while trying to build sandbox for cleanup: network b43b742782192264a37b4a56b387784b137ce3f98e01b791cd555b53df6f3fcf not found
ESC[31mERROESC[0m[0001] failed to delete sandbox 94e581ca6d5e7fe7e1c8a8770e0186839af1eead72f89b53a8332bde9ee2eff5 while trying to cleanup: could not cleanup all the endpoints in container b9a9dbc604fc367c781c538f327f9e5b5f86cb863dab0927e41ab9d47ef80e34 / sandbox 94e581ca6d5e7fe7e1c8a8770e0186839af1eead72f89b53a8332bde9ee2eff5
ESC[31mERROESC[0m[0004] getNetworkFromStore for nid b43b742782192264a37b4a56b387784b137ce3f98e01b791cd555b53df6f3fcf failed while trying to build sandbox for cleanup: network b43b742782192264a37b4a56b387784b137ce3f98e01b791cd555b53df6f3fcf not found
ESC[31mERROESC[0m[0004] failed to delete sandbox 67d282b28fdab4467f84e7109d6484706d5228dc7e59a82d13bb41d248ee6738 while trying to cleanup: could not cleanup all the endpoints in container 560a5995d424233b42b1038ac56dd766daa2e6178118be37b43eae501d3cd437 / sandbox 67d282b28fdab4467f84e7109d6484706d5228dc7e59a82d13bb41d248ee6738
ESC[31mERROESC[0m[0007] getNetworkFromStore for nid 8386ee5c6bb3e40e445d0e64aa9fb50a9933e14d39384442e1e2021ec55f670c failed while trying to build sandbox for cleanup: network 8386ee5c6bb3e40e445d0e64aa9fb50a9933e14d39384442e1e2021ec55f670c not found
ESC[31mERROESC[0m[0007] failed to delete sandbox f22a7ca4a44a8ccf939326de02adcf4c1754c84f390d8eafd1fde75adbf58b4d while trying to cleanup: could not cleanup all the endpoints in container 39cb12f4aa9e8b3f41b80f26a3b393219c35f9192b591e1c371ba9ffa320f407 / sandbox f22a7ca4a44a8ccf939326de02adcf4c1754c84f390d8eafd1fde75adbf58b4d
ESC[31mERROESC[0m[0008] getNetworkFromStore for nid b43b742782192264a37b4a56b387784b137ce3f98e01b791cd555b53df6f3fcf failed while trying to build sandbox for cleanup: network b43b742782192264a37b4a56b387784b137ce3f98e01b791cd555b53df6f3fcf not found
ESC[31mERROESC[0m[0008] failed to delete sandbox 1f1d5cc203108e2ae11395ecdbee79b12cf41f2d9f649b6e487b4777c4275180 while trying to cleanup: could not cleanup all the endpoints in container 89994ecb5802064c91542c47798ab3069a086bfd33880032ab59db09dc4ff33f / sandbox 1f1d5cc203108e2ae11395ecdbee79b12cf41f2d9f649b6e487b4777c4275180
ESC[31mERROESC[0m[0010] getNetworkFromStore for nid 8386ee5c6bb3e40e445d0e64aa9fb50a9933e14d39384442e1e2021ec55f670c failed while trying to build sandbox for cleanup: network 8386ee5c6bb3e40e445d0e64aa9fb50a9933e14d39384442e1e2021ec55f670c not found
ESC[31mERROESC[0m[0010] failed to delete sandbox 26bae7ca4bf4f75f96de3a987b829ff82f5003e33d0bd8ea28acf98da559bced while trying to cleanup: could not cleanup all the endpoints in container def48c66ed8bb5763a21b11174c2190faca629e6c788f70539eb42207330dd4d / sandbox 26bae7ca4bf4f75f96de3a987b829ff82f5003e33d0bd8ea28acf98da559bced
ESC[31mERROESC[0m[0012] nosuchcontainer: no such id: a08f057e92c078427ff7c911f4d97848a71af03e2f3334091c0bab71e3cdb71f
ESC[31mERROESC[0m[0012] nosuchcontainer: no such id: a08f057e92c078427ff7c911f4d97848a71af03e2f3334091c0bab71e3cdb71f
ESC[31mERROESC[0m[0012] nosuchcontainer: no such id: a08f057e92c078427ff7c911f4d97848a71af03e2f3334091c0bab71e3cdb71f
ESC[31mERROESC[0m[0012] failed to umount /var/lib/docker/containers/ed754fe0898f6f5a842720825df5ee47589dad98bda2a61596e8b25e6b2da40e/shm: invalid argument
ESC[31mERROESC[0m[0012] failed to umount /var/lib/docker/containers/ed754fe0898f6f5a842720825df5ee47589dad98bda2a61596e8b25e6b2da40e/mqueue: invalid argument
ESC[31mERROESC[0m[0012] ed754fe0898f6f5a842720825df5ee47589dad98bda2a61596e8b25e6b2da40e: Failed to umount ipc filesystems: failed to cleanup ipc mounts:
ESC[31mERROESC[0m[0012] Failed to start container ed754fe0898f6f5a842720825df5ee47589dad98bda2a61596e8b25e6b2da40e: linknotrunning: Cannot link to a non running container: /callex_centralLog_1 AS /callex_clxNotes_1/callex_centralLog_1
ESC[31mERROESC[0m[0012] failed to cleanup up stale network sandbox for container 89994ecb5802064c91542c47798ab3069a086bfd33880032ab59db09dc4ff33f
ESC[31mERROESC[0m[0012] Error deleting sandbox id 1f1d5cc203108e2ae11395ecdbee79b12cf41f2d9f649b6e487b4777c4275180 for container 89994ecb5802064c91542c47798ab3069a086bfd33880032ab59db09dc4ff33f: could not cleanup all the endpoints in container 89994ecb5802064c91542c47798ab3069a086bfd33880032ab59db09dc4ff33f / sandbox 1f1d5cc203108e2ae11395ecdbee79b12cf41f2d9f649b6e487b4777c4275180
ESC[31mERROESC[0m[0012] failed to umount /var/lib/docker/containers/89994ecb5802064c91542c47798ab3069a086bfd33880032ab59db09dc4ff33f/shm: invalid argument
ESC[31mERROESC[0m[0012] failed to umount /var/lib/docker/containers/89994ecb5802064c91542c47798ab3069a086bfd33880032ab59db09dc4ff33f/mqueue: invalid argument
ESC[31mERROESC[0m[0012] 89994ecb5802064c91542c47798ab3069a086bfd33880032ab59db09dc4ff33f: Failed to umount ipc filesystems: failed to cleanup ipc mounts:
ESC[31mERROESC[0m[0012] Failed to start container 89994ecb5802064c91542c47798ab3069a086bfd33880032ab59db09dc4ff33f: open : no such file or directory
ESC[31mERROESC[0m[0012] failed to cleanup up stale network sandbox for container b9a9dbc604fc367c781c538f327f9e5b5f86cb863dab0927e41ab9d47ef80e34
ESC[31mERROESC[0m[0012] Error deleting sandbox id 94e581ca6d5e7fe7e1c8a8770e0186839af1eead72f89b53a8332bde9ee2eff5 for container b9a9dbc604fc367c781c538f327f9e5b5f86cb863dab0927e41ab9d47ef80e34: could not cleanup all the endpoints in container b9a9dbc604fc367c781c538f327f9e5b5f86cb863dab0927e41ab9d47ef80e34 / sandbox 94e581ca6d5e7fe7e1c8a8770e0186839af1eead72f89b53a8332bde9ee2eff5
ESC[31mERROESC[0m[0012] failed to umount /var/lib/docker/containers/b9a9dbc604fc367c781c538f327f9e5b5f86cb863dab0927e41ab9d47ef80e34/shm: invalid argument
ESC[31mERROESC[0m[0012] failed to umount /var/lib/docker/containers/b9a9dbc604fc367c781c538f327f9e5b5f86cb863dab0927e41ab9d47ef80e34/mqueue: invalid argument
ESC[31mERROESC[0m[0012] b9a9dbc604fc367c781c538f327f9e5b5f86cb863dab0927e41ab9d47ef80e34: Failed to umount ipc filesystems: failed to cleanup ipc mounts:
ESC[31mERROESC[0m[0012] Failed to start container b9a9dbc604fc367c781c538f327f9e5b5f86cb863dab0927e41ab9d47ef80e34: open : no such file or directory
ESC[31mERROESC[0m[0012] failed to cleanup up stale network sandbox for container 560a5995d424233b42b1038ac56dd766daa2e6178118be37b43eae501d3cd437
ESC[31mERROESC[0m[0012] failed to umount /var/lib/docker/containers/a08f057e92c078427ff7c911f4d97848a71af03e2f3334091c0bab71e3cdb71f/shm: invalid argument

after: manual run of docker-compose up - everything is ok.

**Comment by cpuguy83** [comment]
These errors are not indicative of an actual issue, just an unclean shutdown of the daemon.
Do these containers not start up if you try to manually start them?

**Comment by mikechris** [comment]
In version: 1.9.0-rc4, build e6f5a3c  I can manually start containers after reboot.
Problem is with automatic start (after docker daemon start) of containers with policy restart=always/on-failure - after clean reboot (or shutdown). 
Error in my last message is after clean reboot.

**Comment by mikechris** [comment]
@cpuguy83  Sorry I just realized that I copied-pasted  wrong error.
Here is the right one:

ESC[31mERROESC[0m[0003] failed to umount /var/lib/docker/containers/cc1758974685e5013a6b4b0aacf34267c193b5a4cd70396ee825ce6357b61c6c/shm: invalid argument
ESC[31mERROESC[0m[0003] failed to umount /var/lib/docker/containers/cc1758974685e5013a6b4b0aacf34267c193b5a4cd70396ee825ce6357b61c6c/mqueue: invalid argument
ESC[31mERROESC[0m[0003] cc1758974685e5013a6b4b0aacf34267c193b5a4cd70396ee825ce6357b61c6c: Failed to umount ipc filesystems: failed to cleanup ipc mounts:
ESC[31mERROESC[0m[0003] Failed to start container cc1758974685e5013a6b4b0aacf34267c193b5a4cd70396ee825ce6357b61c6c: linknotrunning: Cannot link to a non running container: /callex_clxDbNotelco_1 AS /callex_clxNotes_1/clxDbNotelco_1
ESC[31mERROESC[0m[0003] failed to umount /var/lib/docker/containers/055bdf00d1665d7e7ecd3414072cc27df282ae0ea9bf4c043547f7123a8e0e60/shm: invalid argument
ESC[31mERROESC[0m[0003] failed to umount /var/lib/docker/containers/055bdf00d1665d7e7ecd3414072cc27df282ae0ea9bf4c043547f7123a8e0e60/mqueue: invalid argument
ESC[31mERROESC[0m[0003] 055bdf00d1665d7e7ecd3414072cc27df282ae0ea9bf4c043547f7123a8e0e60: Failed to umount ipc filesystems: failed to cleanup ipc mounts:
ESC[31mERROESC[0m[0003] Failed to start container 055bdf00d1665d7e7ecd3414072cc27df282ae0ea9bf4c043547f7123a8e0e60: linknotrunning: Cannot link to a non running container: /callex_clxDbNotelco_1 AS /callex_provexApi_1/callex_clxDbNotelco_1
ESC[31mERROESC[0m[0004] failed to umount /var/lib/docker/containers/ca9666759ea27b6033153e2023f5d840c73e0efc6a7a204aee2bbb2f9ea14a01/shm: invalid argument
ESC[31mERROESC[0m[0004] failed to umount /var/lib/docker/containers/ca9666759ea27b6033153e2023f5d840c73e0efc6a7a204aee2bbb2f9ea14a01/mqueue: invalid argument
ESC[31mERROESC[0m[0004] ca9666759ea27b6033153e2023f5d840c73e0efc6a7a204aee2bbb2f9ea14a01: Failed to umount ipc filesystems: failed to cleanup ipc mounts:
ESC[31mERROESC[0m[0004] Failed to start container ca9666759ea27b6033153e2023f5d840c73e0efc6a7a204aee2bbb2f9ea14a01: linknotrunning: Cannot link to a non running container: /callex_provexApi_1 AS /callex_clxWWW_1/provexApi
ESC[31mERROESC[0m[0004] failed to umount /var/lib/docker/containers/e1fd2f87b97d9e5a606c304633bb836880754788d5c3c24f25a33c3d4b90b10a/shm: invalid argument
ESC[31mERROESC[0m[0004] failed to umount /var/lib/docker/containers/e1fd2f87b97d9e5a606c304633bb836880754788d5c3c24f25a33c3d4b90b10a/mqueue: invalid argument
ESC[31mERROESC[0m[0004] e1fd2f87b97d9e5a606c304633bb836880754788d5c3c24f25a33c3d4b90b10a: Failed to umount ipc filesystems: failed to cleanup ipc mounts:
ESC[31mERROESC[0m[0004] Failed to start container e1fd2f87b97d9e5a606c304633bb836880754788d5c3c24f25a33c3d4b90b10a: linknotrunning: Cannot link to a non running container: /callex_clxDbNotelco_1 AS /callex_clxPhpPgadmin_1/clxDbNotelco
ESC[31mERROESC[0m[0006] failed to umount /var/lib/docker/containers/6bfe3d316409dd6dfb602e30f1513772ecf559482da9096ea806fb0c6966af1c/shm: invalid argument
ESC[31mERROESC[0m[0006] failed to umount /var/lib/docker/containers/6bfe3d316409dd6dfb602e30f1513772ecf559482da9096ea806fb0c6966af1c/mqueue: invalid argument
ESC[31mERROESC[0m[0006] 6bfe3d316409dd6dfb602e30f1513772ecf559482da9096ea806fb0c6966af1c: Failed to umount ipc filesystems: failed to cleanup ipc mounts:
ESC[31mERROESC[0m[0006] Failed to start container 6bfe3d316409dd6dfb602e30f1513772ecf559482da9096ea806fb0c6966af1c: linknotrunning: Cannot link to a non running container: /callex_provexApi_1 AS /callex_clxLbnoTelco_1/callex_provexApi_1
ESC[31mERROESC[0m[0006] failed to umount /var/lib/docker/containers/42c4bd8690dca34f46385abc3d7ade3053abf8b743b11466f96410a703ad9050/shm: invalid argument
ESC[31mERROESC[0m[0006] failed to umount /var/lib/docker/containers/42c4bd8690dca34f46385abc3d7ade3053abf8b743b11466f96410a703ad9050/mqueue: invalid argument
ESC[31mERROESC[0m[0006] 42c4bd8690dca34f46385abc3d7ade3053abf8b743b11466f96410a703ad9050: Failed to umount ipc filesystems: failed to cleanup ipc mounts:
ESC[31mERROESC[0m[0006] Failed to start container 42c4bd8690dca34f46385abc3d7ade3053abf8b743b11466f96410a703ad9050: linknotrunning: Cannot link to a non running container: /callex_clxNotes_1 AS /callex_clxAgentPanel_1/callex_clxNotes_1

**Comment by thaJeztah** [comment]
@mikechris is this still an issue with docker 1.9.1?

**Comment by mikechris** [comment]
@thaJeztah Problem still exist - new errors:

ERRO[0000] getEndpointFromStore for eid 7b854e06132d9126cfeb0bcec20074bdcf935cc4dce0d23eb7421c82cd2c565d failed while trying to build sandbox for cleanup: could not find endpoint 7b854e06132d9126cfeb0bcec20074bdcf935cc4dce0d23eb7421c82cd2c565d: Key not found in store
ERRO[0001] getEndpointFromStore for eid 67b88696b14a53cd772ea88884b938b144ac5b069c918875abf89ab2bf2d7cf1 failed while trying to build sandbox for cleanup: could not find endpoint 67b88696b14a53cd772ea88884b938b144ac5b069c918875abf89ab2bf2d7cf1: Key not found in store
ERRO[0001] getEndpointFromStore for eid 40451c3bdc771a9457f6aa52894145505742602e31654eb3caea579a264fc351 failed while trying to build sandbox for cleanup: could not find endpoint 40451c3bdc771a9457f6aa52894145505742602e31654eb3caea579a264fc351: Key not found in store
ERRO[0002] getEndpointFromStore for eid 09f62383f4955277cdc349b248ff403b57b676ccf7d26d9fe76b527b592559fc failed while trying to build sandbox for cleanup: could not find endpoint 09f62383f4955277cdc349b248ff403b57b676ccf7d26d9fe76b527b592559fc: Key not found in store
ERRO[0004] getEndpointFromStore for eid 57f906c0fb7fc36d711163582bee4e3edf172077e4bf1ddd62039344ea05450a failed while trying to build sandbox for cleanup: could not find endpoint 57f906c0fb7fc36d711163582bee4e3edf172077e4bf1ddd62039344ea05450a: Key not found in store
ERRO[0004] getEndpointFromStore for eid cb9634db3e408f5a2563d2c7722969382162b57f1be7a52909c4874f79bb7963 failed while trying to build sandbox for cleanup: could not find endpoint cb9634db3e408f5a2563d2c7722969382162b57f1be7a52909c4874f79bb7963: Key not found in store
ERRO[0004] getEndpointFromStore for eid 16ba506f8a5c9201881ecd4098b9da71435a3e850591332104660de2d11f0990 failed while trying to build sandbox for cleanup: could not find endpoint 16ba506f8a5c9201881ecd4098b9da71435a3e850591332104660de2d11f0990: Key not found in store
ERRO[0005] getEndpointFromStore for eid 97123e13c0817a6e477d5b1ef6aae70c8e790ff243c5ace415d070c9595b51b7 failed while trying to build sandbox for cleanup: could not find endpoint 97123e13c0817a6e477d5b1ef6aae70c8e790ff243c5ace415d070c9595b51b7: Key not found in store
ERRO[0005] getEndpointFromStore for eid e64ea0ac71954436f207fb72c232c878f5499130b4c288193cfdd27b06395d66 failed while trying to build sandbox for cleanup: could not find endpoint e64ea0ac71954436f207fb72c232c878f5499130b4c288193cfdd27b06395d66: Key not found in store
ERRO[0007] Failed to start container 9ae0b606c22741736e7b14802a97e8653f25436c583e2f3965b620a9a3ba762a: linknotrunning: Cannot link to a non running container: /callex_clxDbNotelco_1 AS /callex_clxPhpPgadmin_1/clxDbNotelco_1
ERRO[0009] Failed to start container e1ac8d218616185aed34e63b4f4f466338aff35328f092224613287b432e6b52: linknotrunning: Cannot link to a non running container: /callex_centralLog_1 AS /callex_clxNtp_1/centralLog_1
ERRO[0015] Failed to start container 17d8271ad3ab555621d61bdc888f3b29b2d74d14eba53a1da434fc34178f353e: linknotrunning: Cannot link to a non running container: /callex_clxDbNotelco_1 AS /callex_clxNotes_1/callex_clxDbNotelco_1
ERRO[0016] Failed to start container d1eee20283ad917d21584adb6d77ff345e08db490d42f4115690c3bc4d187616: linknotrunning: Cannot link to a non running container: /callex_clxDbNotelco_1 AS /callex_provexApi_1/clxDbNotelco
ERRO[0021] Failed to start container 46a0b12334da154ed5347d8698887539a735fa536fa57c428632f57c8c2c846b: linknotrunning: Cannot link to a non running container: /callex_clxDbNotelco_1 AS /callex_clxWWW_1/callex_clxDbNotelco_1
ERRO[0021] Failed to start container 3c4951da7dc0f739f7d0480bbcffc51c6bf9918b2de96814ce6d47f650cc0f57: linknotrunning: Cannot link to a non running container: /callex_clxWWW_1 AS /callex_clxAgentPanel_1/clxWWW_1
ERRO[0022] Failed to start container b8205929eb7797299cf884db8e5a636d1ca62ee966417e776b8d3609bd7d7629: linknotrunning: Cannot link to a non running container: /callex_clxAgentPanel_1 AS /callex_clxLbnoTelco_1/callex_clxAgentPanel_1

**Comment by thaJeztah** [comment]
Actually, I think this PR https://github.com/docker/docker/pull/18208 (that @cpuguy83 created) is trying to solve this

**Comment by LK4D4** [comment]
@cpuguy83 psst

**Comment by cpuguy83** [comment]
ping @mrjana

**Comment by thaJeztah** [comment]
also related;

- https://github.com/moby/moby/issues/11777

Let me close this ticket for now, as it looks like it went stale, and some fixes were merged (and another ticket is tracking this as well)
